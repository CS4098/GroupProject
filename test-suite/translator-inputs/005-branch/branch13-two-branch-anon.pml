process test{
	sequence seq1 {
		branch {
			action act_1 {
			requires { a }
			provides { b }
			}
			action act_2 {
			requires { b }
			provides { c }
			}
			action act_3 {
			requires { a }
			provides { c }
			}
		}
		branch {
			action act_4 {
			requires { d }
			provides { e }
			}
			action act_5 {
			requires { e }
			provides { f }
			}
		}
	}
}
