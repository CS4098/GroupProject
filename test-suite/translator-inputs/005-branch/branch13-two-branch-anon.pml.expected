bool a;
bool b;
bool c;
bool d;
bool e;
bool f;

active proctype test()
{
	int proc_count_1 = _nr_pr;
	run act_1();
	run act_2();
	run act_3();
	_nr_pr == proc_count_1 ->
	int proc_count_2 = _nr_pr;
	run act_4();
	run act_5();
	_nr_pr == proc_count_2 ->
}

proctype act_1()
{
	a ->
		b = true;
}

proctype act_2()
{
	b ->
		c = true;
}

proctype act_3()
{
	a ->
		c = true;
}

proctype act_4()
{
	d ->
		e = true;
}

proctype act_5()
{
	e ->
		f = true;
}
